File: Backend/src/models/item_tags_to_inventory_item_types.py
# Backend/src/models/item_tags_to_inventory_item_types.py
# Many-to-many relationship between item tags and inventory item types.

# Author: Valentin Haas, 2024

# Library imports
from sqlalchemy import Column, Integer
from ._base import BaseTable


class ItemTagsToInventoryItemTypesTable(BaseTable):
    """Model to track the many-to-many relationship between item tags and inventory item types."""

    __tablename__ = "ItemTagsToInventoryItemTypes"
    ItemTag = Column(Integer, primary_key=True, foreign_key="ItemTags.Id")
    """The item tag of the relationship."""

    InventoryItemType = Column(
        Integer, primary_key=True, foreign_key="InventoryItemTypes.Id"
    )
    """The inventory item type of the relationship."""

File: Backend/src/models/itemtagstoinventoryitemtypes.go
// Backend/src/models/itemtagstoinventoryitemtypes.go
// Many-to-many relationship between item tags and inventory item types.

// Author: Valentin Haas, 2025
package models

import (
	"gorm.io/gorm"
)

// ItemTagsToInventoryItemTypes represents a many-to-many relationship between item tags and inventory item types.
type ItemTagsToInventoryItemType struct {
	gorm.Model                            // Provides ID, CreatedAt, UpdatedAt, DeletedAt fields
	ItemTag             ItemTag           // Referenced ItemTag Object
	ItemTagID           uint              `gorm:"primaryKey; not null"` // The item tag of the relationship.
	InventoryItemType   InventoryItemType // Referenced InventoryItemType Object
	InventoryItemTypeID uint              `gorm:"primaryKey; not null"` // The inventory item type of the relationship.
}
